storm-multilang/dotnet/src/main/dotnet/Dotnet.Storm.Adapter/Components/BaseBolt.cs

```diff
@@ -0,0 +1,133 @@
+            {
+                VerificationResult result = VerifyOutput(stream, tuple);
+
+                if (result.IsError)
+                {
+                    Logger.Error($"Validation failed: {result} in {tuple}.");
+                }
+                else
+                {
+                    BoltTuple message = new BoltTuple()
+                    {
+                        Task = task,
+                        Stream = stream,
+                        Tuple = tuple,
+                        Anchors = anchors,
+                        NeedTaskIds = needTaskIds
+                    };
+
+                    GlobalStorm.Send(message);
+                }
+            }
+        }
+
+        private bool running = true;
+
+        public BaseBolt()
+        {
+            Storm = new LocalStorm();
+        }
+
+        internal override void Start()
+        {
+            while(running)
+            {
+                Message message = GlobalStorm.Receive<ExecuteTuple>();
+                if (message != null)
+                {
+                    // there are only two options: task_ids and tuple to execute
+                    if (message is TaskIdsMessage ids)
+                    {
+                        OnTaskIds(this, new TaskIds(ids));
+                    }
+                    if (message is ExecuteTuple tuple)
+                    {
+                        VerificationResult result = Storm.VerifyInput(tuple.Component, tuple.Stream, tuple.Tuple);
+
+                        if (result.IsError)
+                        {
+                            Logger.Error($"Validation failed: {result} in {tuple}.");
+                        }
+                        else
+                        {
+                            switch (tuple.Stream)
+                            {
+                                case "__heartbeat":
+                                    DoHeartbeat();
+                                    break;
+                                case "__tick":
+                                    DoTick();
+                                    break;
+                                default:
+                                    DoExecute(tuple);
+                                    break;
+                            }
+                        }
+                    }
+                }
+            }
+        }
+
+        private void DoHeartbeat()
+        {
+            Storm.Sync();
+        }
+
+        private void DoTick()
+        {
+            OnTick?.Invoke(this, EventArgs.Empty);
+        }
+
+        private void DoExecute(ExecuteTuple tuple)
+        {
+            try
+            {
+                Execute(new StormTuple(tuple));
+                if (IsGuarantee)
+                {
+                    Storm.Ack(tuple.Id);
+                }
+            }
+            catch (Exception ex)
+            {
+                Logger.Error($"Failed to process tuple. {ex.GetMessageTrace()}");
```

Comment:
ex.ToString() - see my other comments on this - this should be enough - https://stackoverflow.com/questions/4272579/how-to-print-full-stack-trace-in-exception

Reply:
This is not a stack trace. This is all the messages from all inner exceptions.